jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -249,7 +248,24 @@ public Authorization authenticatePush() throws RegistryAuthenticationFailedExcep
 
   @VisibleForTesting
   URL getAuthenticationUrl(String scope) throws MalformedURLException {
-    return new URL(authenticationUrlBase + scope);
+    return new URL(realm + "?" + getUrlEncodedServiceScope(scope));
+  }
+
+  @VisibleForTesting
+  String getAuthenticationFormUrlEncoded(String scope, String token) {
+    return "grant_type=refresh_token&"
+        + getUrlEncodedServiceScope(scope)
+        + "&refresh_token="
+        + token;
+  }
+
+  private String getUrlEncodedServiceScope(String scope) {
```

Comment:
How about `getBasicAuthRequestParameters()`?

Reply:
`getOAuth2AuthRequestBody` since it is not used in url query but in post body?

Reply:
and 
getBasicAuthRequestParameters for replacing existing method `getAuthenticationUrl`?

Reply:
I see it's used only within this class. But no worries. I'll take care of this later.

Reply:
Refactored in #1524.